backend/dao/winesDAO.js

```diff
@@ -1,3 +1,305 @@
+      filters.profile.includes("dry") ||
+      filters.profile.includes("sweet")
+    );
+  }
+  if (typeof filters.profile === "string") {
+    if (filters.profile == "sweet") {
+      //*from docs: to access nested object, dot notation MUST be in quotations
+      query["flavor_profile.sweet"] = { $gte: 1, $lt: 4 };
+    } else if (filters.profile == "dry") {
+      query["flavor_profile.sweet"] = { $gte: 4, $lte: 5 };
+    } else if (filters.profile == "acidic") {
+      query["flavor_profile.bitter"] = { $gte: 3, $lte: 5 };
+    }
+  } else {
+    for (i = 0; i < filters.profile.length; i++) {
+      if (filters.profile[i] == "sweet") {
+        //*from docs: to access nested object, dot notation MUST be in quotations
+        singleFilter["flavor_profile.sweet"] = { $gte: 1, $lt: 4 };
+      } else if (filters.profile[i] == "dry") {
+        singleFilter["flavor_profile.sweet"] = { $gte: 4, $lte: 5 };
+      } else if (filters.profile[i] == "acidic") {
+        singleFilter["flavor_profile.bitter"] = { $gte: 3, $lte: 5 };
+      } else {
+      }
+      filterObject = singleFilter;
+      singleFilter = {};
+      filter_.push(filterObject);
+    }
+    //query.$and = [{ $or: filter_ }];
+    query1 = { $and: filter_ };
+    price_type.push(query1);
+    query1 = {};
+
+    filterObject = {}; //empty again for next query
+    filter_ = [];
+  }
+  return query;
+}
+
+function origineSelectors(filters) {
+  if (filters.origin) {
+    query.country_name = { $eq: filters.origin };
+  }
+}
+
+function typeSelectors(filters) {
+  if (filters.type == "red") {
+    //*text cannot work on a specific property. Had to set up an index in atlas for it to work. $Text searches the db grape_names via the index set up in atlas
+    query.$text = {
+      $search:
+        "Aragonez|Barbera|Blaufränkisch|Cabernet|Carignan|Castelao|Corvina|Dolcetto|Grenache|Malbec|Merlot|Montepulciano|Mourvedre|Nebbiolo|Noir|Pinotage|Primitivo|Shiraz|Shiraz/Syrah|Sangiovese|Tempranillo|Touriga|Trebbiano|Zinfandel|",
+    };
+  } else if (filters.type == "white") {
+    query.$text = {
+      $search:
+        "Arneis|Aligoté|Blanc|Boal Branco|Chardonnay|Chenin|Cortese|Furmint|Gris|Malmsey|Marsanne|Muscadelle|Riesling|Roussanne|Sercial|Terrantez|Trebbiano|Verdelho|",
+    };
+  } else if (filters.type == "rose") {
+    //!
+    query = { $text: { $search: "Blanc" } };
+  } else if (filters.type == "sparkling") {
+    //!Chardonnay|Pinot Noir|Pinot Meunier
+    query = { $text: { $search: "Blanc" } };
+  }
+  return query;
+}
+//? ************************************* */
+//* *********End Declarations************ */
+//* ************************************* */
+
+module.exports = class WinesDAO {
+  static async injectDB(connection) {
+    if (wines) {
+      return;
+    }
+    try {
+      wines = await connection.db(process.env.USER_NS).collection("wines");
+    } catch (e) {
+      console.log(`Unable to establish connection in winesDAO: ${e}`);
+    }
+  }
```

Comment:
This could also be done in the `constructor`, depends if you already have the database connection when instantiate  the DAO.

Reply:
For sure.
So, I just checked how you are using the DAO class, and we can totally skip the `class` concept.
I don't wanna enter in much details  to not overload you, but the `class` concept is part of the Object Oriented Programming (OOP or OO) paradigm: https://www.educative.io/blog/object-oriented-programming.
Paradigm guides completely how you would write and organize code, OO is one them. I grew to dislike OO, it's definitely misused in the industry, so I think we can skip this, but I'll leave as exercise for you read more about it. I should let you know also that there will be many different opinions about coding and specially paradigms which is a high level and abstract concept.
Also one thing that I should tell ahead, JavaScript doesn't support OO properly, it tries to support it because this feature was highly requested by people that were used only to OOP.
Constructor is a concept of OOP, classes too, but I'll recommend you to remove it from the code.
You see the static methods of the DAO classes? You can totally make pure functions of them and remove the class.
And you can export them at the end of the file like:
```js
module.exports = {
  method1,
  method2,
  method3,
}
```